packages/core/src/amazonq/commons/diff.ts

```diff
@@ -32,3 +34,126 @@ export function createAmazonQUri(path: string, tabId: string) {
     // TODO change the featureDevScheme to a more general amazon q scheme
     return vscode.Uri.from({ scheme: featureDevScheme, path, query: `tabID=${tabId}` })
 }
+
+/**
+ * Extracts the file path and selection context from the message.
+ *
+ * @param {any} message - The message object containing the file and selection context.
+ * @returns {Object} - An object with `filePath` and `selection` properties.
+ */
+export function extractFileAndCodeSelectionFromMessage(message: any) {
+    const filePath = message?.context?.activeFileContext?.filePath
+    const selection = message?.context?.focusAreaContext?.selectionInsideExtendedCodeBlock as vscode.Selection
+    return { filePath, selection }
+}
+
+/**
+ * Indents the given code based on the current document's indentation at the selection start.
+ *
+ * @param {any} message - The message object containing the code.
+ * @param {vscode.TextDocument} doc - The VSCode document where the code is applied.
+ * @param {vscode.Selection} selection - The selection range in the document.
+ * @returns {string} - The processed code to be applied to the document.
+ */
+export function getIndentedCode(message: any, doc: vscode.TextDocument, selection: vscode.Selection) {
```

Comment:
can this function reuse https://github.com/aws/aws-toolkit-vscode/blob/5f68d0e255df5b57f87aa2bcd16009667f86b0fb/packages/core/src/shared/utilities/textUtilities.ts#L39 ?

and, this function looks general, not specific to Q. so it can live in the existing `packages/core/src/shared/utilities/textDocumentUtilities.ts` module

Reply:
Done.